BedrockServer.cpp

```diff
@@ -988,6 +998,9 @@ void BedrockServer::runCommand(unique_ptr<BedrockCommand>&& _command, bool isBlo
                         command->complete = true;
                     } else {
                         SINFO("Conflict or state change committing " << command->request.methodLine << " on worker thread.");
+                        if (_enableConflictPageLocks) {
+                            lastConflictPage = db.getLastConflictPage();
```

Comment:
The log line suggests this is reached when there is a conflict OR when the state changed while committing. In either case it would be retried, but if the state changed and there was no conflict, is it still necessary to lock on the conflicting page? If not, perhaps it is not a problem either.

Reply:
It won't matter, If there was no conflict, then this code: https://github.com/Expensify/Bedrock/pull/1521/files#diff-6cca2e6a219aa7f2d6c430b05478aa480eeaafd46c51bbe7e4e43d9ac7a64ce9R283

 won't set `_conflictPage` so lastConflictPage will be `0`, which skips the lock:
 https://github.com/Expensify/Bedrock/pull/1521/files#diff-99a547a03944fca3a89e6b81c3b1034497347dcbc6a2703ec2a80e3fb6815b46R75-R77